.circleci/config.yml

```diff
@@ -0,0 +1,43 @@
+version: 2
+jobs:
+  checkout_cuberite:
+    working_directory: ~/cuberite
+    docker:
+    - image: peterbell10/cuberite-circleci:0.0.3
+    steps:
+    - checkout
+    - run: git submodule update --init
+    - save_cache:
+        key: "cuberite-{{ .Environment.CIRCLE_SHA1 }}"
+        paths:
+        - ~/cuberite
+
+  check_basic_style:
+    working_directory: ~/cuberite/src
+    docker:
+    - image: peterbell10/cuberite-circleci:0.0.3
+    steps:
+    - restore_cache: { keys: [ "cuberite-{{ .Environment.CIRCLE_SHA1 }}" ] }
+    - run: find . -name \*.cpp -or -name \*.h > AllFiles.lst
+    - run: lua CheckBasicStyle.lua
+    - run: cd Bindings && lua CheckBindingsDependencies.lua
+
+  clang-tidy:
+    working_directory: ~/cuberite
+    docker:
+    - image: peterbell10/cuberite-circleci:0.0.3
```

Comment:
You could use YAML-References to reduce the version duplication. See [.travis.yml](https://github.com/cuberite/cuberite/blob/master/.travis.yml#L21) for an example.